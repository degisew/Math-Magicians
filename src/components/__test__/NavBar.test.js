//* eslint-disable */
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

const MockNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

it('Did component has changed?', () => {
  const tree = renderer.create(
    <MockNavbar />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Checking whether its rendering 3 li elementd', () => {
  render(
    <MockNavbar />,
  );
  const listItems = screen.getAllByTestId('li');

  expect(listItems.length).toBe(3);
});
