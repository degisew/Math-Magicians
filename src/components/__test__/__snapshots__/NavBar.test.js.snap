// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Did component has changed? 1`] = `
<header>
  <p
    className="logo"
  >
    Math Magicians
  </p>
  <ul
    className="nav-link"
  >
    <li
      className="li"
      data-testid="li"
    >
      <a
        aria-current="page"
        className="NavLink active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="li"
      data-testid="li"
    >
      <a
        className="NavLink"
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      data-testid="li"
    >
      <a
        className="NavLink"
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</header>
`;
